packages/react/src/animation/TextRevealAnimation/TextRevealAnimation.examples.stories.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react'
+import {Meta} from '@storybook/react'
+import {TextRevealAnimation} from '.'
+import {Testimonial, Box, AnimationProvider, Hero} from '../..'
+
+export default {
+  title: 'Components/TextRevealAnimation/Examples',
+  component: TextRevealAnimation,
+  decorators: [
+    Story => (
+      <AnimationProvider>
+        <Box animate="fade-in">
+          <Story />
+        </Box>
+      </AnimationProvider>
+    ),
+  ],
+} as Meta<typeof TextRevealAnimation>
```

Comment:
Something to discuss in general, but we should probably discourage the use of `as` in the codebase.

In this example, you can put whatever you want into the above object and TS won't complain as we're overriding it with `as`.

If we switched to 

```ts
export const meta: Meta<typeof TextRevealAnimation> = {
    // ...
}
export default meta
```

Then TS will complain if we pass invalid keys

Reply:
Coincidentally I've just run into an example of this with `LogoSuite`.

We're using a type assertion (`as Meta`) in the stories there too and that hides the fact that `default` isn't a valid `marqueeSpeed`. The correct value should have been `normal`.

If I switch it to a type a type annotation (`: Meta`) then that error is correctly highlighted.

https://github.com/primer/brand/pull/691/commits/0d3b6b90c662560a77ce3bcbb1e7b3a841675692

Reply:
I agree with this in principle. One edge case we have is where some stories need to extend the components props with story specific ones. This approach wouldn't directly work there, but we could create an intersection instead? How are you seeing that?

Reply:
Yea that's a good point. I tend to prefer creating a new `ControlsProps` type that works pretty much as you'd expect.

I've got an example of that [here](https://github.com/primer/brand/blob/85bff7aa3d201e2caf5e2cca3c09c4a820e32e71/packages/react/src/LogoSuite/LogoSuite.stories.tsx)

Reply:
You could you that same pattern I mentioned above [here](https://github.com/primer/brand/pull/688/files#diff-5a817314c9e27f624f129949bef8dd3a335fa52920ca35fe94c2b05a47e6799bR38) to remove the `as`, for example.

Reply:
🤔 I see, you're using intersections but manually cherry-picking specific props instead of automatically inheriting and joining them. Is that what's happening? (I could totally be reading this wrong). I can see why we'd need to do that for compound APIs, but it feels a bit duplicative.

If we needed to create story props though, creating a new type makes sense. Could you call it StoryProps though please, we [have something similar for IDE already](https://github.com/primer/brand/blob/main/packages/react/src/IDE/IDE.stories.tsx#L75).

Reply:
I'm creating my own type by `Pick`ing the props I care about from the individual LogoSuite compound components here

```ts
type ControlsProps = Pick<LogoSuiteProps, 'align' | 'hasDivider'> &
  Pick<LogoSuiteHeadingProps, 'visuallyHidden'> &
  Pick<LogoSuiteLogoBarProps, 'marquee' | 'marqueeSpeed' | 'variant'>
```

Which is equivalent to this, just minus the duplication

```ts
type ControlsProps = {
  hasDivider?: boolean,
  align?: 'start' | 'center' | 'justify',
  visuallyHidden?: boolean,
  marquee?: boolean,
  marqueeSpeed?: 'slow' | 'normal' | 'idle',
  variant?: 'muted' | 'emphasis'
} 
```

I then grab those props and put them in the right place in the Playground story

```tsx
export const Playground: StoryFn<ControlsProps> = ({
  hasDivider,
  align,
  visuallyHidden,
  marquee,
  marqueeSpeed,
  variant,
}) => (
  <LogoSuite align={align} hasDivider={hasDivider}>
    <LogoSuite.Heading visuallyHidden={visuallyHidden}>Heading</LogoSuite.Heading>
    <LogoSuite.Description>Body text maximus ligula felis, non egestas dolor rutrum vel.</LogoSuite.Description>
    <LogoSuite.Logobar variant={variant} marquee={marquee} marqueeSpeed={marqueeSpeed}>
      {logos.slice(0, 5)}
    </LogoSuite.Logobar>
  </LogoSuite>
)
```